tools/monorepo-utils/src/core/github/repo.ts

```diff
@@ -167,3 +170,51 @@ export const createPullRequest = async ( options: {
 	//@ts-ignore There is a type mismatch between the graphql schema and the response. pullRequest.data.head.repo.has_discussions is a boolean, but the graphql schema doesn't have that field.
 	return pullRequest.data;
 };
+
+/**
+ * Get a pull request from Github.
+ *
+ * @param {Object} options
+ * @param {string} options.owner repository owner.
+ * @param {string} options.name  repository name.
+ * @param          prNumber      pull request number.
+ * @return {Promise<object>}     pull request data.
+ */
+export const getPullRequest = async (
+	options: { owner: string; name: string },
+	prNumber: string
+): Promise< PullRequestEndpointResponse[ 'data' ] > => {
+	const { owner, name } = options;
+	const pr = await octokitWithAuth().request(
+		'GET /repos/{owner}/{repo}/pulls/{pull_number}',
+		{
+			owner,
+			repo: name,
+			pull_number: Number( prNumber ),
+		}
+	);
+
+	//@ts-ignore
```

Comment:
This confused me as I couldn't see any 'why' for this. I then found this comment in a different function. Is this the rationale? 

```suggestion
	//@ts-ignore There is a type mismatch between the graphql schema and the response. pullRequest.data.head.repo.has_discussions is a boolean, but the graphql schema doesn't have that field.
```

When I removed the `@ts-ignore` in VS Code it did not complain. How / when would I be able to trigger it complaining -- only when building TS as part of a build step? Or is my IDE setup possibly incomplete here?

Reply:
When I remove it in my IDE, I get the following which is different than your suggestion above.

<img width="829" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/9622ac72-901e-4a8a-84ec-3b6430cacba8">

I have made sure I'm on all the latest Octokit packages and this still persists. I don't know why there is an error, so I didn't add an explanation. Open to suggestions here

Reply:
Weird and unfortunate, but nothing we can or should solve as part of this PR. 

I'd add a comment giving maybe the last sentence of the error message and basically something like "we don't know why this error comes up, we'll try to solve it in a future iteration" -- just so that someone not super familiar with the code (e.g. us in six months) will understand our why.